### PR TITLE
devops: add Trace Viewer deployment

### LINKED ISSUE #10015: [internal] Auto deploy the trace viewer

### PR DESCRIPTION
**Pull request by mxschmitt**
Fixes #10015

### CODE REVIEW

File: utils/build/deploy-trace-viewer.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ENVIRONMENT="$1"
+
+# 1. Install dependencies and build the Trace Viewer
+npm ci
+npm run build
+
+# 2. Configure Git and clone the Trace Viewer repository
+git config --global user.name github-actions
+git config --global user.email 41898282+github-actions[bot]@users.noreply.github.com
+git clone https://$GH_SERVICE_ACCOUNT_TOKEN@github.com/microsoft/playwright-trace.git playwright-trace
```
Comment (aslushnikov) [inline utils/build/deploy-trace-viewer.sh:32]:
```suggestion
git clone "https://${GH_SERVICE_ACCOUNT_TOKEN}@github.com/microsoft/playwright-trace.git" playwright-trace
```

File: utils/build/deploy-trace-viewer.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+# 3. Copy the built Trace Viewer to the repository
```
Comment (aslushnikov) [inline utils/build/deploy-trace-viewer.sh:34]: Looks like build step is missing!

File: utils/build/deploy-trace-viewer.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+if [[ "${ENVIRONMENT}" == "production" ]]; then
+  cp -r packages/playwright-core/lib/webpack/traceViewer/* playwright-trace/
+  echo "Updated production version"
+else
```
Comment (aslushnikov) [inline utils/build/deploy-trace-viewer.sh:38]:
```suggestion
elif [[ "${ENVIRONMENT}" == "canary" ]]; then
```

File: utils/build/deploy-trace-viewer.sh [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  cp -r packages/playwright-core/lib/webpack/traceViewer/ playwright-trace/next/
+  echo "Updated canary version"
+fi
```
Comment (aslushnikov) [inline utils/build/deploy-trace-viewer.sh:41]:
```suggestion
else
  echo "ERROR: unknown option - ${ENVIRONMENT}"
  exit 1
fi
```

File: utils/build/deploy-trace-viewer.sh (hunk 2 above)
Comment (mxschmitt) [reply utils/build/deploy-trace-viewer.sh:34]: see step 1

**Review by aslushnikov — APPROVED** [review]

**Pull request merged**